src/librustc_trans/back/link.rs

```diff
@@ -938,8 +938,10 @@ fn add_upstream_rust_crates(cmd: &mut Linker,
             Linkage::NotLinked |
             Linkage::IncludedFromDylib => {}
             Linkage::Static => {
+                let is_a_no_builtins_crate =
+                    attr::contains_name(&sess.cstore.crate_attrs(cnum), "no_builtins");
```

Comment:
I've seen an `CrateTranslation` struct that contains a `no_builtins` field. I'll see if I can pass it around in these contexts.